src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -408,15 +408,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             {
                 auto lock = _terminal->LockForWriting();
                 _terminal->UpdateSelection(updateSlnParams->first, updateSlnParams->second, modifiers);
-                _renderer->TriggerSelection();
+                _updateSelection(false);
                 return true;
             }
 
             // GH#8791 - don't dismiss selection if Windows key was also pressed as a key-combination.
             if (!modifiers.IsWinPressed())
             {
                 _terminal->ClearSelection();
-                _renderer->TriggerSelection();
+                _updateSelection(true);

```

Comment:
At least from the first four calls to `_updateSelection`, it seems like we could use some known state ("is there a selection?") to determine whether to show/clear the markers.